src/proxy/app/db_manager.py

```diff
@@ -1,61 +1,149 @@
+                self.db_config.get_connection_string(), command_timeout=60
             )
+            self.logging.info("Connection pool created")
+            self.pool_timestamp = datetime.now()
         except asyncpg.exceptions.PostgresError as error:
             self.logging.error("Postgres error: %s", str(error))
             raise HTTPException(
-                status_code=501, detail=f"Postgres error opening pool exp {str(error)}"
+                status_code=503, detail=f"Postgres error opening pool exp {str(error)}"
             ) from error
 
         except Exception as exception:
             self.logging.error("Error: %s", str(exception))
             raise HTTPException(
-                status_code=501, detail=f"Postgres error opening pool exp {str(exception)}"
+                status_code=503, detail=f"Postgres error opening pool exp {str(exception)}"
             ) from exception
 
     async def close_pool(self):
         """close database pool"""
-        print("Closing connection pool")
+        self.logging.info("Closing connection pool")
         try:
-            await self.app.pool.close()
+            await asyncio.wait_for(self.db_pool.close(), timeout=60)
+        except TimeoutError as error:
+            self.logging.error("Postgres close timeout exception: %s", str(error))
+            self.logging.error("Postgres pool terminated: %s", str(error))
+            self.db_pool.terminate()
         except asyncpg.exceptions.PostgresError as error:
             self.logging.error("Postgres error: %s", str(error))
             raise HTTPException(
-                status_code=501, detail=f"Postgres error closing pool {str(error)}"
+                status_code=503, detail=f"Postgres error closing pool {str(error)}"
             ) from error
-
         except Exception as exception:
             self.logging.error("Error: %s", str(exception))
             raise HTTPException(
-                status_code=501, detail=f"Postgres error closing pool {str(exception)}"
+                status_code=503, detail=f"Postgres exception closing pool {str(exception)}"
             ) from exception
 
-    async def get_connection(self):
-        """connect to database"""
-        return self.app.pool
-
     def get_postgres_encryption_key(self):
         """get postgres encryption key"""
-        return self.postgres_encryption_key
+        return self.db_config.encryption_key
+
+    async def __aenter__(self):
+        """Get a connection from the pool"""
+        # If the pool is older than RECYCLE_TIME_SECONDS, generate new connection string
+        if (datetime.now() - self.pool_timestamp).total_seconds() > RECYCLE_TIME_SECONDS:
+            self.logging.info("Renewing connection pool connection string")
+            # generate a new connection string for new connections to use in the pool
+            self.db_pool.set_connect_args(
+                dsn=self.db_config.get_connection_string(), command_timeout=60
```

Comment:
Do you have to re-specify command_timeout here? It doesnt inherit the original value?